### PR TITLE
Use barrels to reduce imports

### PR DESCRIPTION
**Pull request by deepu105**
Part of #3664 

Based on angular2 guideline suggestion to use barrels to group imports per module.

### CODE REVIEW

File: generators/client-2/templates/src/main/webapp/app/_app.ng2module.ts [hunk 1]
```diff
@@ -3,25 +3,18 @@ import { BrowserModule } from '@angular/platform-browser';
...
-import { HomeComponent } from './home/home.component';
-import { NavbarComponent } from './layouts/navbar/navbar.component';
-import { FooterComponent } from './layouts/footer/footer.component';
-import { ErrorComponent } from './layouts/error/error.component';
+import { appState } from './app.state';
+import { HomeComponent, homeState } from './home';
+import { NavbarComponent, FooterComponent, ErrorComponent, errorState, accessdeniedState } from './layouts';
 <%_ if (enableTranslation){ _%>
-import { ActiveMenuDirective } from './layouts/navbar/active-menu.directive';
+import { ActiveMenuDirective } from './layouts';
```
Comment (sendilkumarn) [inline generators/client-2/templates/src/main/webapp/app/_app.ng2module.ts]: can it be added above ?

File: generators/client-2/templates/src/main/webapp/app/account/settings/_settings.component.ts [hunk 2]
```diff
@@ -1,9 +1,9 @@
 import { Component, Inject, OnInit } from '@angular/core';
 <%_ if (enableTranslation){ _%>
-import { <%=jhiPrefixCapitalized%>LanguageService } from '../../components/language/language.service';
+import { <%=jhiPrefixCapitalized%>LanguageService } from '../../components';
 <%_ } _%>
-import { Principal } from '../../components/auth/principal.service';
-import { Account } from '../../components/auth/account.service';
+import { Principal } from '../../components';
+import { Account } from '../../components';
```
Comment (sendilkumarn) [inline generators/client-2/templates/src/main/webapp/app/account/settings/_settings.component.ts]: Couple these together please

File: generators/client-2/templates/src/main/webapp/app/_app.ng2module.ts (hunk 1 above)
Comment (deepu105) [inline generators/client-2/templates/src/main/webapp/app/_app.ng2module.ts]: It looked ugly 😉 with the ejs, but you are right it should be above

File: generators/client-2/templates/src/main/webapp/app/layouts/navbar/_navbar.component.ts [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
-import { ProfileService } from '../../components/profiles/profile.service';
+import { ProfileService } from '../../components';
 <%_ if (enableTranslation){ _%>
-import { <%=jhiPrefixCapitalized%>LanguageService } from '../../components/language/language.service';
+import { <%=jhiPrefixCapitalized%>LanguageService } from '../../components';
 <%_ } _%>
-import { Principal } from '../../components/auth/principal.service';
-import { AuthService } from '../../components/auth/auth.service';
+import { Principal } from '../../components';
+import { AuthService } from '../../components';
```
Comment (sendilkumarn) [inline generators/client-2/templates/src/main/webapp/app/layouts/navbar/_navbar.component.ts]: couple here too

File: generators/client-2/templates/src/main/webapp/app/layouts/navbar/_navbar.component.ts [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
```
Comment (sendilkumarn) [inline generators/client-2/templates/src/main/webapp/app/layouts/navbar/_navbar.component.ts]: couple here too

**Comment by sendilkumarn** [comment]
@deepu105 it was one of my `the earliest` ask to add barrels and broccoli 😉 

It started with `barrels`, hopefully we can see `broccoli` soon if it is not dead.

**Comment by sendilkumarn** [comment]
Inbtn super cool 👏  LGTM apart from the minor stuff.

File: generators/client-2/templates/src/main/webapp/app/account/settings/_settings.component.ts (hunk 2 above)
Comment (deepu105) [inline generators/client-2/templates/src/main/webapp/app/account/settings/_settings.component.ts]: 👍

**Comment by deepu105** [comment]
barrels may be I missed I wanted to do it always. Brocolli not so much, coz just no benefit over gulp

**Pull request merged**